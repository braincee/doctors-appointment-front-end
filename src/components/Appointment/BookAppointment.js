import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { userAppointment } from '../../redux/appointment/appointmentReducer';
import './bookappointment.css';

const AppointmentForm = () => {
  const [city, setCity] = useState('');
  const [appointmenTime, setTime] = useState('');
  const [doctorId, setId] = useState('');

  const doctors = useSelector((state) => state.doctorReducer);
  const dispatch = useDispatch();

  const handleChange = (event) => {
    const getTarget = event.target;
    const inputValue = getTarget.value;
    switch (getTarget.name) {
      case 'city':
        setCity(inputValue);
        return city;
      case 'appointmenTime':
        setTime(inputValue);
        return appointmenTime;
      case 'doctorId':
        setId(inputValue);
        return doctorId;
      default:
        return 'yes';
    }
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    dispatch(userAppointment(city, appointmenTime, doctorId));
    event.target.reset();
  };

  return (
    <section className="appointment-section">
      <div className="appointment-container">
        <h3 className="appointment-title">Book an Appointment</h3>
        <form onSubmit={handleSubmit} className="appointment-form">
          <input
            type="text"
            className="form-control"
            id="city"
            name="city"
            value={city}
            onChange={handleChange}
            placeholder="Enter city"
            required
          />
          <input
            className="form-control"
            id="datetime"
            type="datetime-local"
            name="appointmenTime"
            value={appointmenTime}
            min="2022-08-17T00:00"
            max="2050-08-17T00:00"
            onChange={handleChange}
            required
          />
          <select value={doctorId} onChange={handleChange} name="doctorId" className="select">
            <option value="">Select your doctor</option>
            {doctors.length > 0 && doctors.map((doctor) => (
              <option key={doctor.id} value={doctor.id}>
                {doctor.name}
              </option>
            ))}

          </select>
          <input type="submit" value="Book Appointment" className="button1 btn btn-secondary" />
        </form>
      </div>
    </section>
  );
};

export default AppointmentForm;
