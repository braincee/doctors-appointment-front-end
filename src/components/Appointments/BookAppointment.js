import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import userAppointment from '../../redux/appointments/actions/appointmentActions';
import './bookappointment.css';

const BookAppointment = () => {
  const [city, setCity] = useState('');
  const [appointmenTime, setTime] = useState('');
  const [doctorId, setId] = useState('');

  const doctors = useSelector((state) => state.doctors);
  const user = useSelector((state) => state.user);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleChange = (event) => {
    const getTarget = event.target;
    const inputvalue = getTarget.value;
    switch (getTarget.name) {
      case 'city':
        setCity(inputvalue);
        return city;
      case 'appointmenTime':
        setTime(inputvalue);
        return appointmenTime;
      case 'doctorId':
        setId(inputvalue);
        return doctorId;
      default:
        return 'yes';
    }
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const appointment = {
      city,
      appointment_time: appointmenTime,
    };

    dispatch(userAppointment(appointment, user.id, doctorId));
    setCity('');
    setTime('');
    setId();
    setTimeout(() => { navigate('/my_appointments'); }, 1000);
  };

  return (
    <section className="appointments-section">
      <div className="appointments-container">
        <h3 className="appointments-title">Book an Appointment</h3>
        <form onSubmit={handleSubmit} className="appointments-form">
          <input
            type="text"
            className="form-control"
            id="city"
            name="city"
            value={city}
            onChange={handleChange}
            placeholder="Enter city"
            required
          />
          <input
            className="form-control"
            id="datetime"
            type="datetime-local"
            name="appointmenTime"
            value={appointmenTime}
            min="2022-08-17T00:00"
            max="2050-08-17T00:00"
            onChange={handleChange}
            required
          />
          <select id="doctor" value={doctorId} onChange={handleChange} name="doctorId" className="select">
            <option value="">Select your doctor</option>
            {doctors.length > 0 ? doctors.map((doctor) => (
              <option key={doctor.id} value={doctor.id}>
                {doctor.name}
              </option>
            )) : 'No Doctors available'}
          </select>
          <input type="submit" value="Book Appointment" className="button1 btn btn-secondary" />
        </form>
      </div>
    </section>
  );
};

export default BookAppointment;
