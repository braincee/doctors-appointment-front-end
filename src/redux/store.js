import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';




const initialState = {};
const middleware = [thunk];

const store = createStore(initialState, (applyMiddleware(...middleware)));