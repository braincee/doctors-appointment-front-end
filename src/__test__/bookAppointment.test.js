import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import BookAppointment from '../components/Appointments/BookAppointment';
import store from '../redux/configureStore';

describe('Book Appointment form tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <BookAppointment />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Book Appointment page has a "Book Appointment" button', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <BookAppointment />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Book Appointment')).toBeTruthy();
  });
});
