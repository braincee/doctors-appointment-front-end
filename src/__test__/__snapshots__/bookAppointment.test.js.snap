// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Appointment form tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="appointments-section"
      >
        <div
          class="appointments-container"
        >
          <h3
            class="appointments-title"
          >
            Book an Appointment
          </h3>
          <form
            class="appointments-form"
          >
            <input
              class="form-control"
              id="city"
              name="city"
              placeholder="Enter city"
              required=""
              type="text"
              value=""
            />
            <input
              class="form-control"
              id="datetime"
              max="2050-08-17T00:00"
              min="2022-08-17T00:00"
              name="appointmenTime"
              required=""
              type="datetime-local"
              value=""
            />
            <select
              class="select"
              id="doctor"
              name="doctorId"
            >
              <option
                value=""
              >
                Select your doctor
              </option>
              No Doctors available
            </select>
            <input
              class="button1 btn btn-secondary"
              type="submit"
              value="Book Appointment"
            />
          </form>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="appointments-section"
    >
      <div
        class="appointments-container"
      >
        <h3
          class="appointments-title"
        >
          Book an Appointment
        </h3>
        <form
          class="appointments-form"
        >
          <input
            class="form-control"
            id="city"
            name="city"
            placeholder="Enter city"
            required=""
            type="text"
            value=""
          />
          <input
            class="form-control"
            id="datetime"
            max="2050-08-17T00:00"
            min="2022-08-17T00:00"
            name="appointmenTime"
            required=""
            type="datetime-local"
            value=""
          />
          <select
            class="select"
            id="doctor"
            name="doctorId"
          >
            <option
              value=""
            >
              Select your doctor
            </option>
            No Doctors available
          </select>
          <input
            class="button1 btn btn-secondary"
            type="submit"
            value="Book Appointment"
          />
        </form>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
